doc/user/metrics/README.md

```diff
@@ -13,43 +25,50 @@ By default, the metrics are served on `0.0.0.0:8383/metrics`. To modify the port
 #### Usage:
 
 ```go
-    import(
-        "context"
-
+    import (
+        ...
         "github.com/operator-framework/operator-sdk/pkg/metrics"
-        "sigs.k8s.io/controller-runtime/pkg/manager"
-        "k8s.io/api/core/v1"
+        v1 "k8s.io/api/core/v1"
         "k8s.io/apimachinery/pkg/util/intstr"
+        "sigs.k8s.io/controller-runtime/pkg/manager"
+        ...        
     )
 
+    // Change below variables to serve metrics on different host or port.
+    var (
+        metricsHost               = "0.0.0.0"
+        metricsPort         int32 = 8383
+        operatorMetricsPort int32 = 8686
+    )
+    
+    ...
+
     func main() {
 
         ...
 
-        // Change the below variables to serve metrics on different host or port.
-        var metricsHost = "0.0.0.0"
-        var metricsPort int32 = 8383
-
-        // Pass metrics address to controller-runtime manager
-        mgr, err := manager.New(cfg, manager.Options{
-            Namespace:          namespace,
-            MetricsBindAddress: fmt.Sprintf("%s:%d", metricsHost, metricsPort),
-        })
+       // Pass metrics address to controller-runtime manager
+       // Create a new Cmd to provide shared dependencies and start components
+       mgr, err := manager.New(cfg, manager.Options{
+          Namespace: namespace,
+          MetricsBindAddress: fmt.Sprintf("%s:%d", metricsHost, metricsPort),
+       })
 
         ...
 
         // Add to the below struct any other metrics ports you want to expose.
-	    servicePorts := []v1.ServicePort{
-		    {Port: metricsPort, Name: metrics.OperatorPortName, Protocol: v1.ProtocolTCP, TargetPort: intstr.IntOrString{Type: intstr.Int, IntVal: metricsPort}},
-	    }
-
-        // Create Service object to expose the metrics port.
-        _, err = metrics.CreateMetricsService(context.TODO(), cfg, servicePorts)
-        if err != nil {
-            // handle error
-        }
-
-        ...
+      	servicePorts := []v1.ServicePort{
+      		{Port: metricsPort, Name: metrics.OperatorPortName, Protocol: v1.ProtocolTCP, TargetPort: intstr.IntOrString{Type: intstr.Int, IntVal: metricsPort}},
+      		{Port: operatorMetricsPort, Name: metrics.CRPortName, Protocol: v1.ProtocolTCP, TargetPort: intstr.IntOrString{Type: intstr.Int, IntVal: operatorMetricsPort}},
+      	}
+
+      	// Create Service object to expose the metrics port(s).
+      	service, err := metrics.CreateMetricsService(ctx, cfg, servicePorts)
+      	if err != nil {
+      		// handle error here
+      	}
+      
+      	...
```

Comment:
This snippet is about creating the service that exposes the metrics. So maybe we separate this out into a section about exposing metrics and include the snippet that sets up the ServiceMonitor?

With that in mind, we might need to tweak the doc organization to use the following sections:
1. Rename `General metrics` -> `Configuring controller-runtime metrics`
2. Rename `Custom resource specific metrics` -> `Configuring metrics about custom resources`
3. Move and rename `Expose Custom Metrics` -> Subsection of `Configuring controller-runtime metrics` called `Defining custom metrics`
4. New section called `Exposing metrics`, that includes snippets for service creation and service monitor creation, and includes the `Garbage collection` section since that is referring to the service that gets created.

Would that make this doc flow a little better? WDYT?